rules/no-array-fill-with-reference-type.js

```diff
@@ -0,0 +1,141 @@
+// @ts-check
+const MESSAGE_ID_ERROR = 'no-array-fill-with-reference-type/error';
+const messages = {
+	[MESSAGE_ID_ERROR]: 'Avoid using `{{actual}}` with reference type{{type}}. Use `Array.from({ ... }, () => { ... })` instead to ensure independent instances.',
+};
+
+const debugging = false;
+const log = (...arguments_) => debugging && console.log(...arguments_);
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => ({
+	CallExpression(node) {
+		const isArrayFill = node.callee.type === 'MemberExpression'
+			&& ((node.callee.object.callee?.name === 'Array') || (context.sourceCode.getText(node.callee.object.callee) === 'Array.from'))
+			&& node.callee.property.name === 'fill'
+			&& node.arguments.length > 0;
+
+		log('isArrayFill:', isArrayFill);
+
+		if (!isArrayFill) {
+			return;
+		}
+
+		const fillArgument = node.arguments[0];
+		log('fillArgument:', fillArgument);
+
+		if (!isReferenceType(fillArgument, context)) {
+			return;
+		}
+
+		let type = '';
+		switch (fillArgument.type) {
+			case 'ObjectExpression': {
+				type = 'Object';
+				break;
+			}
+
+			case 'ArrayExpression': {
+				type = 'Array';
+				break;
+			}
+
+			case 'NewExpression': {
+				type = `new ${fillArgument.callee.name}()`;
+				break;
+			}
+
+			case 'FunctionExpression':
+			case 'ArrowFunctionExpression': {
+				type = 'Function';
+				break;
+			}
+
+			default: {
+				if (fillArgument.type === 'Literal' && fillArgument.regex) {
+					type = 'RegExp';
+				} else if (fillArgument.type === 'Identifier') {
+					type = `variable (${fillArgument.name})`;
+				}
+			}
+		}
+
+		const actual = context.sourceCode.getText(node.callee.object.callee) === 'Array.from' ? 'Array.from().fill()' : 'Array.fill()';
+
+		return {
+			node,
```

Comment:
```suggestion
			node: node.arguments[0],
```
Let's report on `node.arguments[0]` instead.